foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/EnableHabushuBuildCacheMigration.java

```diff
@@ -0,0 +1,116 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+
+
+/**
+ * This migration adds a new build directory tag to the build node of the project's pom.xml file.
```

Comment:
S: especially with XLST (or regex, or similar), explaining exactly what is happening in human-readable language is typically very valuable.  For instance - this would be a lot more directly understandable by someone if we said something like: ensuring that habushu modules leverage the dist folder for their output rather than target to conform to typical python conventions and ensure the build cache can find and replace those artifacts via default configuration.

I might argue that the class name should be aligned to say exactly what it does too.

Reply:
Agreed WRT a more explanatory comment.  For the class name, I could go either way-- The functional impact of the migration is that we're making sure that the build cache works as expected for Habushu-packaged modules, as reflected in the current name.  If we decide we'd rather talk about the specifics of the migration rather than the functional outcome, I'm fine with that, but I don't think we have a set norm at this time.  I would cite other migrations like `TiltfileMigration`, and `DockerPipInstallMigration`.

Reply:
Seems like `EnableDistOutputFolderForHabushuBuildCacheMigration` or similar is a lot more clear and hard to argue with IMO.

If I had done PRs on those, I'd have had a similar comment.

Reply:
Fair enough.  Done.